### PR TITLE
Alelo: Adding homologation changes

### PR DESCRIPTION
**Pull request by Heavyblade**
## Summary:

Adding changes to properly use the encryption key UUID on transactions, update types for amount and expiration year.

Adding a separate file for the remote tests that represents the tests cases used for certification

## Unit Test:

Finished in 39.980662 seconds.
5343 tests, 76567 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## Remote Test:

test/remote/gateways/remote_alelo_test.rb
Finished in 35.552773 seconds.
17 tests, 43 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

test/remote/gateways/remote_alelo_test_certification.rb Finished in 49.461157 seconds.
14 tests, 38 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## RuboCop:
750 files inspected, no offenses detected

### CODE REVIEW

File: lib/active_merchant/billing/gateways/alelo.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module ActiveMerchant #:nodoc:
   module Billing #:nodoc:
     class AleloGateway < Gateway
       self.test_url = 'https://sandbox-api.alelo.com.br/alelo/sandbox/'
-      self.live_url = 'https://desenvolvedor.alelo.com.br'
+      self.live_url = 'https://api.alelo.com.br/alelo/prd/'
```
Comment (Agajanth) [inline lib/active_merchant/billing/gateways/alelo.rb:7]: should we include a variable with the homologation URL?  like self.pre_live

File: lib/active_merchant/billing/gateways/alelo.rb [hunk 2]
```diff
@@ -82,12 +82,12 @@ def add_geolocation(post, options)
         })
       end
 
-      def add_payment(post, payment, options)
+      def add_payment(post, payment)
         post.merge!({
           cardNumber: payment.number,
           cardholderName: payment.name,
           expirationMonth: payment.month,
```
Comment (Agajanth) [inline lib/active_merchant/billing/gateways/alelo.rb:89]: shouldn't the month also be an integer?

**Review by Agajanth — APPROVED** [review]

File: lib/active_merchant/billing/gateways/alelo.rb (hunk 1 above)
Comment (curiousepic) [reply lib/active_merchant/billing/gateways/alelo.rb:7]: That would be good - we do this for some other gateways like `litle`

File: test/unit/gateways/alelo_test.rb [hunk 3]
```diff
@@ -70,12 +71,12 @@ def test_successful_purchase_with_provided_credentials
       decrypted = JOSE::JWE.block_decrypt(secret_key, JSON.parse(data)['token']).first
       request = JSON.parse(decrypted, symbolize_names: true)
 
-      assert_equal @options[:order_id], request[:request_id]
-      assert_equal '1.00', request[:amount]
+      assert_equal @options[:order_id], request[:requestId]
+      assert_equal 1.0, request[:amount]
```
Comment (curiousepic) [inline test/unit/gateways/alelo_test.rb:75]: Is this really correct? are we sure only `.to_f` is sufficient for correct formatting? I wouldn't expect `1.0` to be proper.

File: lib/active_merchant/billing/gateways/alelo.rb (hunk 2 above)
Comment (Heavyblade) [reply lib/active_merchant/billing/gateways/alelo.rb:92]: The Month is indeed an integer

File: lib/active_merchant/billing/gateways/alelo.rb (hunk 1 above)
Comment (Heavyblade) [reply lib/active_merchant/billing/gateways/alelo.rb:9]: Done

File: test/unit/gateways/alelo_test.rb (hunk 3 above)
Comment (Heavyblade) [reply test/unit/gateways/alelo_test.rb:75]: well, the money method applies the proper format but it returns '1.00' and Alelo needs float number, documentation says 'string' but based on the homologation work and the API example, I'm sending the return of amount method as a float number.

capture___sandbox-desenvolvedor_alelo_com_br_br

Comment (Heavyblade) [reply test/unit/gateways/alelo_test.rb:75]: mmm that should not be correct, I tested on a random local test with an amount of 100 that end up being translated as 1.0
tmux

Comment (curiousepic) [reply test/unit/gateways/alelo_test.rb:75]: Here is what I'm seeing:
Screen Shot 2022-09-26 at 4 19 57 PM

Comment (curiousepic) [reply test/unit/gateways/alelo_test.rb:75]: I do see `1.0` in the request, but that is what I get for the response.

Comment (Heavyblade) [reply test/unit/gateways/alelo_test.rb:75]: Ahhh the beauty of Alelo's Sandbox @curiousepic , no matter what you send in the body, using the uuid '5314 ...' you are always going to get that response:


Screen Shot 2022-09-26 at 3 28 15 PM

Comment (curiousepic) [reply test/unit/gateways/alelo_test.rb:75]: Aah, it looks like the sandbox always returns that amount 😩

**Review by curiousepic — APPROVED** [review]
👍  Thanks for the tweaks

**Comment by curiousepic** [comment]
Closed with 68a7f5b0ae6ededb0a4ec127b3eaaecfb4700722

**Pull request closed without merging**